client/state/plugins/installed/actions.js

```diff
@@ -617,34 +617,59 @@ export function removePlugin( siteId, plugin ) {
 	};
 }
 
-export function fetchPlugins( siteIds ) {
+export function receiveSitePlugins( siteId, plugins ) {
+	return {
+		type: PLUGINS_RECEIVE,
+		data: plugins,
+		siteId,
+	};
+}
+
+export function fetchSitePlugins( siteId ) {
 	return ( dispatch, getState ) => {
-		return siteIds.map( ( siteId ) => {
-			const defaultAction = {
-				siteId,
-			};
-			dispatch( { ...defaultAction, type: PLUGINS_REQUEST } );
+		const state = getState();
+		const site = getSite( state, siteId );
+		const defaultAction = {
+			siteId,
+		};
+		dispatch( { ...defaultAction, type: PLUGINS_REQUEST } );
+
+		const afterFetchCallback = ( error, data ) => {
+			// @TODO: Remove when this flux action is completely reduxified
+			Dispatcher.handleServerAction( {
+				type: 'RECEIVE_PLUGINS',
+				action: 'RECEIVE_PLUGINS',
+				site,
+				data,
+				error,
+			} );
+		};
 
-			const receivePluginsDispatchSuccess = ( data ) => {
-				dispatch( { ...defaultAction, type: PLUGINS_RECEIVE, data: data.plugins } );
-				dispatch( { ...defaultAction, type: PLUGINS_REQUEST_SUCCESS } );
+		const receivePluginsDispatchSuccess = ( data ) => {
+			dispatch( receiveSitePlugins( siteId, data.plugins ) );
+			afterFetchCallback( undefined, data );
+			dispatch( { ...defaultAction, type: PLUGINS_REQUEST_SUCCESS } );
 
-				data.plugins.map( ( plugin ) => {
-					if ( plugin.update && plugin.autoupdate ) {
-						updatePlugin( siteId, plugin )( dispatch, getState );
-					}
-				} );
-			};
+			data.plugins.map( ( plugin ) => {
+				if ( plugin.update && plugin.autoupdate ) {
+					updatePlugin( siteId, plugin )( dispatch, getState );
+				}
+			} );
+		};
 
-			const receivePluginsDispatchFail = ( error ) => {
-				dispatch( { ...defaultAction, type: PLUGINS_REQUEST_FAILURE, error } );
-			};
+		const receivePluginsDispatchFail = ( error ) => {
+			dispatch( { ...defaultAction, type: PLUGINS_REQUEST_FAILURE, error } );
+			afterFetchCallback( error, undefined );
+		};
 
-			return wpcom
-				.site( siteId )
-				.pluginsList()
-				.then( receivePluginsDispatchSuccess )
-				.catch( receivePluginsDispatchFail );
-		} );
+		return wpcom
+			.site( siteId )
+			.pluginsList()
```

Comment:
Previously there was a distinction between `.pluginsList` and `.wpcomPluginsList`, is this not necessary anymore? Or is it that we only handle plugins in Calypso for jetpack sites anyway?

Reply:
Good question! Yep, it's no longer necessary; we're [redirecting to the WP.org plugin browser](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/main.jsx#L70) for non Jetpack sites, so the old WP.com plugins list is no longer used anywhere anymore.